src/factory-reset/factory-reset-tool.c

```diff
@@ -0,0 +1,295 @@
+
+                default:
+                        assert_not_reached();
+                }
+
+        return 1;
+}
+
+static int verb_status(int argc, char *argv[], void *userdata) {
+        enum {
+                /* Report current mode also as via exit status, but only return a subset of states */
+                EXIT_OFF     = EXIT_SUCCESS,
+                EXIT_ON      = 10,
+                EXIT_PENDING = 11,
+        };
+
+        FactoryResetMode f = factory_reset_mode();
+        if (f < 0)
+                return log_error_errno(f, "Failed to determine factory reset mode: %m");
+
+        if (!arg_quiet)
+                puts(factory_reset_mode_to_string(f));
+
+        return f == FACTORY_RESET_ON ? EXIT_ON :
+                f == FACTORY_RESET_PENDING ? EXIT_PENDING : EXIT_OFF;
+}
+
+static int verb_request(int argc, char *argv[], void *userdata) {
+        int r;
+
+        FactoryResetMode f = factory_reset_mode();
+        if (f < 0)
+                return log_error_errno(f, "Failed to determine current factory reset mode: %m");
+        if (f == FACTORY_RESET_ON)
+                return log_error_errno(SYNTHETIC_ERRNO(EBUSY), "System currently in factory reset mode, refusing to request another one.");
+        if (f == FACTORY_RESET_PENDING) {
+                log_info("Factory request already requested, skipping.");
+                return 0;
+        }
+
+        if (!is_efi_boot())
+                return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP),
+                                       "Not an EFI boot, requesting factory reset via EFI variable not supported.");
+
+        _cleanup_free_ char *id = NULL, *image_id = NULL, *version_id = NULL, *image_version = NULL;
+        r = parse_os_release(
+                        /* root= */ NULL,
+                        "ID", &id,
+                        "IMAGE_ID", &image_id,
+                        "VERSION_ID", &version_id,
+                        "IMAGE_VERSION", &image_version);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse os-release: %m");
+
+        if (!id)
+                return log_error_errno(SYNTHETIC_ERRNO(EBADR), "os-release data lacks ID= field, refusing.");
+
+        sd_id128_t boot_id;
+        r = sd_id128_get_boot(&boot_id);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get boot ID: %m");
+
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        r = sd_json_buildo(
+                        &v,
+                        SD_JSON_BUILD_PAIR_STRING("osReleaseId", id),
+                        SD_JSON_BUILD_PAIR_CONDITION(!!version_id, "osReleaseVersionId", SD_JSON_BUILD_STRING(version_id)),
+                        SD_JSON_BUILD_PAIR_CONDITION(!!image_id, "osReleaseImageId", SD_JSON_BUILD_STRING(image_id)),
+                        SD_JSON_BUILD_PAIR_CONDITION(!!image_version, "osReleaseImageVersion", SD_JSON_BUILD_STRING(image_version)),
```

Comment:
ImageVersion is not parsed by factory-reset.c . Is it intentional, may be for future extension? Or simply forgotten?